DNN Platform/Library/Common/Utilities/FileExtensionWhitelist.cs

```diff
@@ -141,5 +141,19 @@ private IEnumerable<string> NormalizeExtensions(IEnumerable<string> additionalEx
         {
             return additionalExtensions.Select(ext => (ext.StartsWith(".") ? ext : "." + ext).ToLowerInvariant());
         }
+
+        public FileExtensionWhitelist RestrictBy(FileExtensionWhitelist parentList)
+        {
+            var filter = parentList.AllowedExtensions.ToList();
```

Comment:
Alternatively, since we're just using contains, using a set may be more performant:
```suggestion
            var filter = parentList.AllowedExtensions.ToHashSet();
```